### ISSUE TITLE
Binding a ui.date element to a ui.input element will trigger on_change on the ui.input

### ISSUE THREAD

**Issue by ghbm-itk** [zauberzeug/nicegui#2236]
### Description

When binding a date element to a input element it will immediatly trigger the input element's on_change function.
It doesn't seem to happen with other elements. I have tried a few as seen in the code snippet.

This is a problem because my on_change function is dependent on other elements that doesn't exists at the binding time.

```python
from nicegui import ui

def change():
    print("Change!")

input_1 = ui.input(on_change=change)

print("Bind input")
input_2 = ui.input()
input_1.bind_value(input_2)

print("Bind date")
date_input = ui.date()
input_1.bind_value(date_input)

print("Bind time")
time_input = ui.time()
input_1.bind_value(time_input)

print("Bind number")
number_input = ui.number()
input_1.bind_value(number_input)

print("Done")
```

**Comment by falkoschindler** [comment]
Hi @ghbm-itk, thanks for reporting this strange behavior!

The reason seems to be that `ui.input`'s default value is an empty string `''`, while it's `None` for the other elements. When you use two-way binding, the calling object gets updated if the values differ. In this case `input1` updates from `''` to `None`.

Here is an overview of default values for several value elements:

- `ui.color_input`: `str = ''`
- `ui.editor`: `str = ''`
- `ui.input`: `str = ''`
- `ui.textarea`: `str = ''`
- `ui.date`: `Optional[str] = None`
- `ui.slider`: `Optional[float] = None`
- `ui.splitter`: `Optional[float] = 50`
- `ui.time`: `Optional[str] = None`

So when an element is supposed to contain string values, its default is an empty string.
And when it should contain numbers or strings in a specific format, its default is `None` or a specific number.

One could argue that all value elements should be equally initialized with `None`. But for some elements like `ui.menu` or `ui.dialog` a value of `None` doesn't make much sense, since the element is either opened or closed. Besides that changing the default values would be a breaking change and is probably not the intention of this issue.

Long story short: I would argue that - given the elements' default values - the behavior is correct.

**Comment by ghbm-itk** [comment]
That's fair.

Just as a note to anyone who might stumble upon this. The problem can be solved by setting the value of the input element to None at initialization. `ui.input(value=None)`. Just be sure that all bound elements have the same value before binding.
